packages/material/src/complex/MaterialEnumArrayControl.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  and,
+  ControlProps,
+  hasType,
+  Paths,
+  RankedTester,
+  rankWith,
+  schemaMatches,
+  schemaSubPathMatches,
+  uiTypeIs
+} from '@jsonforms/core';
+import { MuiCheckbox } from '../mui-controls';
+import {
+  FormControl,
+  FormControlLabel,
+  FormGroup,
+  FormHelperText,
+  Hidden
+} from '@material-ui/core';
+import { startCase } from 'lodash';
+import { isEmpty } from 'lodash';
+import React, { useMemo, useRef } from 'react';
+
+const toUiData = (options: any, data: any) => {
+  const result = Array(options.length);
+  options.forEach((option: any, index: number) => {
+    result[index] = data !== undefined && data.includes(option.const);
+  });
+  return result;
+};
+
+export const MaterialEnumArrayControl = ({
+  schema,
+  visible,
+  errors,
+  path,
+  data,
+  handleChange,
+  ...otherProps
+}: ControlProps) => {
+  const options = useMemo(() => (schema.items as any).anyOf, [schema]);
+  const uiData = useMemo(() => toUiData(options, data), [data, options]);
+
+  const onChange = (option: string, value: any) => {
+    const newData = data ? [...data] : [];
+    if (value) {
+      // option was added
+      newData.push(option);
+    } else {
+      // option was removed
+      const indexInData = newData.indexOf(option);
+      newData.splice(indexInData, 1);
+    }
+
+    handleChange(path, newData);
+  };
+  // we are using a ref for the onChange function because
+  // changing only the handleChange prop of MuiCheckbox does not trigger an update
+  const onChangeRef = useRef(onChange);
+  onChangeRef.current = onChange;
+  return (
+    <Hidden xlUp={!visible}>
+      <FormControl component='fieldset'>
+        <FormGroup row>
+          {uiData.map((value: any, index: number) => {
+            const option = options[index];
+            const optionPath = Paths.compose(path, `${index}`);
+            return (
+              <FormControlLabel
+                id={option.const}
```

Comment:
Careful: The value of `const` can be anything! It doesn't need to be a string, but could for example be an `object`, `array`, etc. Please see `mapStateToOneOfEnumProps` where we already had to handle the case.